pkg/sdk/errors.go

```diff
@@ -53,3 +54,85 @@ func decodeDriverError(err error) error {
 
 	return err
 }
+
+type Error struct {
+	file         string
+	line         int
+	message      string
+	nestedErrors []error
+}
+
+func (e *Error) Error() string {
+	builder := new(strings.Builder)
+	writeTree(e, builder, 0)
+	return builder.String()
+}
+
+// NewError creates new sdk.Error with information like filename or line number (depending on where NewError was called)
+func NewError(message string) error {
+	return newSDKError(message, 2)
+}
+
+// JoinErrors returns an error that wraps the given errors. Any nil error values are discarded.
+// JoinErrors returns nil if errs contains no non-nil values, otherwise returns sdk.Error with errs as its nested errors
+func JoinErrors(errs ...error) error {
+	notNilErrs := make([]error, 0)
+	for _, err := range errs {
+		if err != nil {
+			notNilErrs = append(notNilErrs, err)
+		}
+	}
+	if len(notNilErrs) == 0 {
+		return nil
+	}
+	err := newSDKError("joined error", 2)
+	err.nestedErrors = notNilErrs
+	return err
+}
+
+func newSDKError(message string, skip int, nested ...error) *Error {
```

Comment:
instead of calling this SDK, should we call it "SnowflakeError"?

Reply:
Yeah, forgot about that one. This is pretty much a constructor for sdk.Error, but it was meant to be used only inside errors.go. it's kinda low level in a sense that "skip" parameter can be confusing. I would prefer not to export this and call it "newError" or how you proposed, but in unexported version "snowflakeError" or "newSnowflakeError". I'll rename it to "newError" and comment it that it's supposed to be used only inside this file. Sounds good ?